app/asset-manifest.js

```diff
@@ -1,13 +1,22 @@
 import environment from './config/environment';
 
-const metaName = environment.modulePrefix + '/asset-manifest';
+const modulePrefix = environment.modulePrefix;
+const metaName = `${modulePrefix}/asset-manifest`;
+const nodeName = `${modulePrefix}/node-asset-manifest`;
+
 let config = {};
 
 try {
-  const rawConfig = document.querySelector('meta[name="' + metaName + '"]').getAttribute('content');
-  config = JSON.parse(unescape(rawConfig));
+  // If we have a Node version of the asset manifest, use that for FastBoot and
+  // similar environments.
+  if (require.has(nodeName)) {
```

Comment:
As it stands we're effectively guaranteeing that you always use the node module since it will exist (we insert it every time). In that case why use the meta at all? The primary reason to have the meta is to allow the manifest to change independently from `app.js`. That all of this gets bundled for the node environment makes the browser environment less optimal.